manifests/redhat.pp

```diff
@@ -14,12 +14,21 @@
 
   if $manage_repo {
 
-    $keys = [
+    $keys_src = [
         'https://keys.datadoghq.com/DATADOG_RPM_KEY_CURRENT.public',
-        'https://keys.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public',
         'https://keys.datadoghq.com/DATADOG_RPM_KEY_FD4BF915.public',
         'https://keys.datadoghq.com/DATADOG_RPM_KEY_B01082D3.public',
+        'https://keys.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public',
     ]
+    if $agent_version =~ /([0-9]+:)?([0-9]+)\.([0-9]+)\.([0-9]+)((?:~|-)[^0-9\s-]+[^-\s]*)?(?:-([0-9]+))?/ or $agent_version == 'latest' {
```

Comment:
Since this is the same condition with the same regex as https://github.com/DataDog/puppet-datadog-agent/blob/5b111de56b0706a76d6fbfba218040953b487280/manifests/redhat.pp#L23 (as well as some other places, I think)
Is there any chance that this could be moved to a single place where we can reuse it from (as well as making it easier to read by giving it a name)? Or pre-parsing things into variables ahead of time.

Reply:
I'll see to make a class to decompose the agent version